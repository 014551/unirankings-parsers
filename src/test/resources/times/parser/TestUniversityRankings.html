<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Test title</title>
</head>
<body>
<table id="datatable-1"
       class="table rankingsRegWall-processed wur-hash-processed wur-cols-processed wur-pagelen-processed dataTable no-footer rank-only usr-processed table-row-click-processed scores"
       data-module="rankings_table" role="grid" aria-describedby="datatable-1_info">
    <thead>
    <tr role="row">
        <th class="rank sorting_asc" tabindex="0" aria-controls="datatable-1" rowspan="1" colspan="1"
            aria-label="Rank: activate to sort column ascending">Rank
        </th>
        <th class="name namesearch sorting" tabindex="0" aria-controls="datatable-1" rowspan="1" colspan="1"
            aria-label="NameCountry/Region: activate to sort column ascending">Name<br>Country/Region
        </th>
        <th class="scores overall-score sorting" title="The overall score" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="Overall: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">Overall</div>
        </th>
        <th class="scores teaching-score sorting" title="" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="Teaching: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">Teaching</div>
        </th>
        <th class="scores research-score sorting" title="" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="Research: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">Research</div>
        </th>
        <th class="scores citations-score sorting" title="" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="Citations: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">Citations</div>
        </th>
        <th class="scores industry_income-score sorting" title="" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="Industry Income: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">Industry Income</div>
        </th>
        <th class="scores international_outlook-score sorting" title="" data-placement="top" data-toggle="tooltip"
            data-container="body" tabindex="0" aria-controls="datatable-1"
            aria-label="International Outlook: activate to sort column ascending" rowspan="1" colspan="1">
            <div class="rotate">International Outlook</div>
        </th>
    </tr>
    </thead>
    <tbody>
    <tr role="row" class="odd row-1 js-row" data-nid="468">
        <td class="rank sorting_1 sorting_2">1</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/university-oxford" data-position="title"
                                        data-mz="">University of Oxford</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnxQAAS">United Kingdom</a></span></div>
                <a href="/cn/world-university-rankings/university-oxford?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">94.3</td>
        <td class=" scores teaching-score">86.7</td>
        <td class=" scores research-score">99.5</td>
        <td class=" scores citations-score">99.1</td>
        <td class=" scores industry_income-score">63.7</td>
        <td class=" scores international_outlook-score">95.0</td>
    </tr>
    <tr role="row" class="even row-2 js-row" data-nid="470">
        <td class="rank sorting_1 sorting_2">2</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/university-cambridge" data-position="title"
                                        data-mz="">University of Cambridge</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnxQAAS">United Kingdom</a></span></div>
                <a href="/cn/world-university-rankings/university-cambridge?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">93.2</td>
        <td class=" scores teaching-score">87.8</td>
        <td class=" scores research-score">97.8</td>
        <td class=" scores citations-score">97.5</td>
        <td class=" scores industry_income-score">51.5</td>
        <td class=" scores international_outlook-score">93.0</td>
    </tr>
    <tr role="row" class="odd row-3 js-row" data-nid="128779">
        <td class="rank sorting_1 sorting_2">=3</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/california-institute-technology-caltech"
                                        data-position="title" data-mz="">California Institute of Technology</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/california-institute-technology-caltech?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">93.0</td>
        <td class=" scores teaching-score">90.3</td>
        <td class=" scores research-score">97.5</td>
        <td class=" scores citations-score">99.5</td>
        <td class=" scores industry_income-score">92.6</td>
        <td class=" scores international_outlook-score">59.7</td>
    </tr>
    <tr role="row" class="even row-4 js-row" data-nid="467">
        <td class="rank sorting_1 sorting_2">=3</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/stanford-university" data-position="title"
                                        data-mz="">Stanford University</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/stanford-university?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">93.0</td>
        <td class=" scores teaching-score">89.1</td>
        <td class=" scores research-score">96.7</td>
        <td class=" scores citations-score">99.9</td>
        <td class=" scores industry_income-score">60.5</td>
        <td class=" scores international_outlook-score">77.6</td>
    </tr>
    <tr role="row" class="odd row-5 js-row" data-nid="471">
        <td class="rank sorting_1 sorting_2">5</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/massachusetts-institute-technology"
                                        data-position="title" data-mz="">Massachusetts Institute of Technology</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/massachusetts-institute-technology?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">92.5</td>
        <td class=" scores teaching-score">87.3</td>
        <td class=" scores research-score">91.9</td>
        <td class=" scores citations-score">100.0</td>
        <td class=" scores industry_income-score">88.4</td>
        <td class=" scores international_outlook-score">87.6</td>
    </tr>
    <tr role="row" class="even row-6 js-row" data-nid="466">
        <td class="rank sorting_1 sorting_2">6</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/harvard-university" data-position="title"
                                        data-mz="">Harvard University</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/harvard-university?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">91.8</td>
        <td class=" scores teaching-score">84.2</td>
        <td class=" scores research-score">98.4</td>
        <td class=" scores citations-score">99.7</td>
        <td class=" scores industry_income-score">46.4</td>
        <td class=" scores international_outlook-score">79.7</td>
    </tr>
    <tr role="row" class="odd row-7 js-row" data-nid="469">
        <td class="rank sorting_1 sorting_2">7</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/princeton-university" data-position="title"
                                        data-mz="">Princeton University</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/princeton-university?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">91.1</td>
        <td class=" scores teaching-score">85.7</td>
        <td class=" scores research-score">93.9</td>
        <td class=" scores citations-score">99.6</td>
        <td class=" scores industry_income-score">58.0</td>
        <td class=" scores international_outlook-score">78.7</td>
    </tr>
    <tr role="row" class="even row-8 js-row" data-nid="472">
        <td class="rank sorting_1 sorting_2">8</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/imperial-college-london" data-position="title"
                                        data-mz="">Imperial College London</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnxQAAS">United Kingdom</a></span></div>
                <a href="/cn/world-university-rankings/imperial-college-london?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">89.2</td>
        <td class=" scores teaching-score">81.7</td>
        <td class=" scores research-score">88.7</td>
        <td class=" scores citations-score">96.7</td>
        <td class=" scores industry_income-score">71.6</td>
        <td class=" scores international_outlook-score">96.6</td>
    </tr>
    <tr role="row" class="odd row-9 js-row" data-nid="473">
        <td class="rank sorting_1 sorting_2">9</td>
        <td class=" name namesearch"><a class="ranking-institution-title"
                                        href="/world-university-rankings/university-chicago" data-position="title"
                                        data-mz="">The University of Chicago</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnyEAAS">United States</a></span></div>
                <a href="/cn/world-university-rankings/university-chicago?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">88.6</td>
        <td class=" scores teaching-score">85.3</td>
        <td class=" scores research-score">90.1</td>
        <td class=" scores citations-score">99.4</td>
        <td class=" scores industry_income-score">39.8</td>
        <td class=" scores international_outlook-score">69.6</td>
    </tr>
    <tr role="row" class="even row-10 js-row" data-nid="479">
        <td class="rank sorting_1 sorting_2">=10</td>
        <td class=" name namesearch"><a class="ranking-institution-title" href="/world-university-rankings/eth-zurich"
                                        data-position="title" data-mz="">ETH Zurich</a>
            <div class="location-courses js-location-courses">
                <div class="location"><span><a href="/location/a4zw0000000GnxqAAC">Switzerland</a></span></div>
                <a href="/cn/world-university-rankings/eth-zurich?utm_medium=thewebsite&amp;utm_campaign=cta-link&amp;utm_source=rankings#institution-courses-pane"
                   class="btn btn-default btn-sm courses-button js-courses-button" data-position="uni-apply" data-mz=""
                   data-original-title="" title=""><span>课程</span></a></div>
        </td>
        <td class=" scores overall-score">87.7</td>
        <td class=" scores teaching-score">76.4</td>
        <td class=" scores research-score">92.0</td>
        <td class=" scores citations-score">94.3</td>
        <td class=" scores industry_income-score">60.3</td>
        <td class=" scores international_outlook-score">98.1</td>
    </tr>
    </tbody>
</table>
</body>
</html>