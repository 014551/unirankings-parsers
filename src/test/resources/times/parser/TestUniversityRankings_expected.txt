"0";"1";"University of Oxford";"/world-university-rankings/university-oxford";"United Kingdom";"94.3";"86.7";"99.5";"99.1";"63.7";"95.0"
"1";"2";"University of Cambridge";"/world-university-rankings/university-cambridge";"United Kingdom";"93.2";"87.8";"97.8";"97.5";"51.5";"93.0"
"2";"=3";"California Institute of Technology";"/world-university-rankings/california-institute-technology-caltech";"United States";"93.0";"90.3";"97.5";"99.5";"92.6";"59.7"
"3";"=3";"Stanford University";"/world-university-rankings/stanford-university";"United States";"93.0";"89.1";"96.7";"99.9";"60.5";"77.6"
"4";"5";"Massachusetts Institute of Technology";"/world-university-rankings/massachusetts-institute-technology";"United States";"92.5";"87.3";"91.9";"100.0";"88.4";"87.6"
"5";"6";"Harvard University";"/world-university-rankings/harvard-university";"United States";"91.8";"84.2";"98.4";"99.7";"46.4";"79.7"
"6";"7";"Princeton University";"/world-university-rankings/princeton-university";"United States";"91.1";"85.7";"93.9";"99.6";"58.0";"78.7"
"7";"8";"Imperial College London";"/world-university-rankings/imperial-college-london";"United Kingdom";"89.2";"81.7";"88.7";"96.7";"71.6";"96.6"
"8";"9";"The University of Chicago";"/world-university-rankings/university-chicago";"United States";"88.6";"85.3";"90.1";"99.4";"39.8";"69.6"
"9";"=10";"ETH Zurich";"/world-university-rankings/eth-zurich";"Switzerland";"87.7";"76.4";"92.0";"94.3";"60.3";"98.1"